File: tcp_udp_experiment/chat_app/websocket_exp/sock_server.js
const ws = require("ws")
const wss = new ws.WebSocketServer({port:8765})

wss.on("connection", function connection(ws){
    ws.on("error", console.error);
    ws.on("message", function message(data){
        console.log(`received: ${data}`)
    })
    ws.send("connected")
})

File: tcp_udp_experiment/chat_app/websocket_exp/sock_server.py
import asyncio
from websockets.asyncio.server import serve 

def sync_echo(websocket):
    for message in websocket:
        websocket.send(message)
        print(message)

async def echo(websocket):
    async for message in websocket:
        await websocket.send(message)
        print(message)

async def main():
    async with serve(echo, "127.0.0.1", port=8765) as server:
        await server.serve_forever()

asyncio.run(main())